src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -562,14 +562,16 @@ public virtual void Clear()
         /// </summary>
         /// <param name="newPath">The path to save to</param>
         /// <param name="core"></param>
-        public virtual bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore)
+        /// <param name="isBackup">Indicates whether saved workspace is backup or not. If it's not backup,
+        /// we should add it to recent files. Otherwise leave it.</param>
+        public virtual bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore, bool isBackup = false)
         {
             if (String.IsNullOrEmpty(newPath)) return false;
 
             Log(String.Format(Resources.SavingInProgress, newPath));
             try
             {
-                if (SaveInternal(newPath, runtimeCore))
+                if (SaveInternal(newPath, runtimeCore) && !isBackup)
```

Comment:
Although this works, I'd have hoped we can continue to send this notification to whoever that is listening. The fact is that the `WorkspaceModel` is saved, and whoever listens to it needs to know. A slightly better way to do that of course is to introduce a `WorkspaceSavedEventArgs` that carries both `WorkspaceModel` and `isBackup` data, and then the listener can decide what to do with it (e.g. whether or not we are saving the file for backup, etc.)